### PR TITLE
Got Multiple Styles to work in MPS

### PR DESCRIPTION
**Pull request by abhishekpratapa**
- I was doing bad things.

### CODE REVIEW

**Comment by abhishekpratapa** [comment]
Fixes #2723

**Comment by abhishekpratapa** [comment]
## Overview
Multiple styles wasn't working in TuriCreate because of this silly problem.

## Issue
- The instance norm layers which determine the stylistic feature of an image are supplied by a class called the `InstanceNormDataLoader`. This data loader takes care of all of the `beta` and `gamma` weights during inference as well as the `velocity` and `momentum` buffers during training.

- Two methods exist that need to be overridden on this data loader: the `gamma` and the `beta` method.

- The instanceNorm layer in mps only supports one style meaning one `gamma` and `beta` pair.

- To circumvent this limitation, in the InstanceNormDataLoader class we created `NSArrays` to store the different `gamma` and `beta` values.

- We used an indexing variable `styleIndex` to select the different `gamma` and `beta` values for different styles.

- To export the weights after training I added a concat to the `gamma` and `beta` methods so that all of the `gamma` and `beta` weights for all styles are concatenated into one contagious block of `NSData`

- This worked during export because the part of the style transfer application TuriCreate owned was aware that these were concatenated `gamma` and `beta` values.

- this didn't work during `predict` because these concatenated values were truncated by MPS since they had no awareness that these exported `gamma` and `beta` values were for multiple styles.

- Since the gamma and beta values were contagious the truncation happened to the `gamma` and `beta` values for the first index only.

## The Solution

- Separating the methods that exported the `gamma` and `beta` weights and the methods that gave the current `gamma` and current `beta` weights for the selected style, solved this issue.

- calling `gammaWeights` and `betaWeights` gave the callee the concatenated values of `gamma` and `beta` weights for all styles

- calling `gamma` and `beta` only gave the `gamma` and `beta` values for the current selected style.

In conclusion `MPS` was happy and `TuriCreate` exported all the weights it needed for multi style style-transfer.
